rholang/src/main/k/rholang/rho-syntax.k

```diff
@@ -321,4 +321,108 @@ syntax HigherMatchCasePat  ::= HigherProcPat "=>" "{" HigherProcPat "}" [binder]
+              | "[" String "][" HigherPat "][" HigherPat "]"
+              // Nil
+              | "[" "]"
+              // Hashed new HigherNames
+              | "[" Int "]"
+              | "[" HigherProcPat "][" HigherProcPat "]"
+
+// Syntax which is needed for various matching operations
+syntax CanBeStdMatched ::= Id
+              | "intostdmatchform(" HigherPat ")"
+              | "intostdtfmatchform(" HigherPat ")"
+              | String StringlessMatchFormSyntax
+              | Ground
+              | HigherNameNoVars
+              | KResult
+
+// *******************************************************************
+//        Syntax specifically for the MATCHING-WITH-PAR module
+// *******************************************************************
+
+// Some syntax specifically for matching with "|"
+
+// When matching against the pattern "par" the matching process becomes a little more complicated
+syntax PatternMatchingCells ::=
+   "[" String "][" HigherProcPat "][" HigherProcPat "][" HigherProcPat "][" Int "]["
+                                                                           HigherProcPat "]{"Set"}"
+ | "[" HigherProcPat "][" HigherProcPat "][" HigherProcPat "][" HigherProcPat "][" HigherProcPat "]"
+     // For variable matching at the end
+ | "[" HigherProcPat "][" HigherProcPat "][" HigherProcPat "][" HigherProcPat "]["
+                                                                  HigherProcPat "]{"HigherProcPat"}"
+
+// *******************************************************************
+//           Syntax specifically for the FREE module
+// *******************************************************************
+
+// First we construct the syntax for our nodes
+syntax Ints ::= List{ Int, ","}
+
+syntax Set ::= "List2Set(" List ")"
+syntax List ::= "Set2List(" Set ")"
+
+syntax HigherProcsOrHigherNames ::= List{HigherProcOrHigherName, ","}
+              | HigherProcs
+              | HigherNames
+
+// Sets and Lists
+syntax SetOrList ::= Set
+              | List
+
+syntax NodeSyntax ::=
+            // For normal nodes, the syntax is
+            //[ Id ][ Proc/Name ][ Parent ][ Children ]{ Set of free vars }[ Node constructed ]
+             "[" Int "][" HigherPat "][" Exp "][" List "]{" SetOrList "}[" Bool "]"
+            // Nodes corresponding to binders (bnodes) are given the syntax
+            // [ Id ][ Parent ][ Plus ID ; Plus Process/Name ][ Minus ID ; Minus Process/Name
+            //                                        ]{ Set of free vars }[ Node constructed (T/F)]
+          | "[" Int "][" Exp "][" Int ";" HigherPats "][" List ";" HigherPats "]{" SetOrList "}[" Bool "]"
+            // We need to make sure that a binder doesn't try to bind two variables twice. To do
+            // this, we put a special marker on those elements corresponding to a binder
+          //| "[" Int "][" HigherPat "][" Exp "][" List "]{" SetOrList "}{" List "}{""}[" Bool "]"
+
 endmodule
```

Comment:
Each syntax block that was removed is put together here, under appropriate headings.